file/file.go

```diff
@@ -615,34 +615,61 @@ func (r *Router) GetSharedFiles(c *gin.Context, queryAppID string) {
 
 	// Return a page of all shared files' permissions which belong to the user,
 	// filtered by appID. If queryAppID = "", it will not filter by apps
-	permissions, err := r.permissionClient().GetUserPermissions(
-		c.Request.Context(),
-		&ppb.GetUserPermissionsRequest{
-			UserID:   reqUser.ID,
-			AppID:    queryAppID,
-			PageNum:  pageNum,
-			PageSize: pageSize,
-			IsShared: true},
-	)
-
+	filters := &ppb.GetUserPermissionsRequest{
+		UserID:   reqUser.ID,
+		AppID:    queryAppID,
+		PageNum:  pageNum,
+		PageSize: pageSize,
+		IsShared: true}
+
+	permissionsRes, err := r.permissionClient().GetUserPermissions(c.Request.Context(), filters)
 	if err != nil {
 		httpStatusCode := gwruntime.HTTPStatusFromCode(status.Code(err))
 		loggermiddleware.LogError(r.logger, c.AbortWithError(httpStatusCode, err))
 
 		return
 	}
+	permissions := permissionsRes.GetPermissions()
 
-	// Make an empty slice of files
-	filesSuccesful := make([]*GetFileByIDResponse, 0, len(permissions.GetPermissions()))
-	filesFailed := make([]string, 0, len(permissions.GetPermissions()))
+	fileIds := make([]string, 0, len(permissions))
+	for _, permission := range permissions {
+		fileIds = append(fileIds, permission.GetFileID())
+	}
 
-	for _, permission := range permissions.GetPermissions() {
-		file, err := r.fileClient().GetFileByID(c.Request.Context(), &fpb.GetByFileByIDRequest{Id: permission.GetFileID()})
-		if err != nil {
-			loggermiddleware.LogError(r.logger, fmt.Errorf("failed fetching file %v: %v", permission.GetFileID(), err))
+	filesRes, err := r.fileClient().GetFilesByIDs(c.Request.Context(), &fpb.GetByFilesByIDsRequest{Ids: fileIds})
+	if err != nil {
+		httpStatusCode := gwruntime.HTTPStatusFromCode(status.Code(err))
+		loggermiddleware.LogError(r.logger, c.AbortWithError(httpStatusCode, err))
+		return
+	}
+	files := filesRes.GetFiles()
+
+	filesSuccesful := make([]*GetFileByIDResponse, 0, len(permissions))
+	filesFailed := make([]string, 0, len(permissions))
+
+	type permissionAndFile struct {
+		permission *ppb.GetUserPermissionsResponse_FileRole
+		file       *fpb.File
+	}
+	permissionsAndFiles := make([]*permissionAndFile, 0, len(permissions))
+
+	for _, permission := range permissions {
+		isTheFileExists := false
```

Comment:
change variable name to: "doesFileExist"

```suggestion
		doesFileExist := false
```